packages/provider/src/KnownRelaysManager.ts

```diff
@@ -143,11 +149,39 @@ export class KnownRelaysManager {
     return relayEventParts.flat()
   }
 
-  async _fetchRecentlyActiveRelayManagers (): Promise<Set<Address>> {
+  async _getLatestRelayHubEvents (eventNames?: string[], managers?: Set<Address>): Promise<EventData[]> {
     const toBlock = await this.contractInteractor.getBlockNumber()
-    const fromBlock = Math.max(0, toBlock - this.config.relayLookupWindowBlocks)
 
-    const relayEvents: any[] = await this.getPastEventsForHub(fromBlock, toBlock)
+    let relayEvents: EventData[]
+    if (toBlock - this.latestScannedBlock < 10) {
+      // re-use the same cached events for ~2 minutes.
+      relayEvents = this.cachedLatestEvents
+    } else {
+      const fromBlock = Math.max(0, toBlock - this.config.relayLookupWindowBlocks)
+
+      relayEvents = await this.getPastEventsForHub(fromBlock, toBlock)
+      this.cachedLatestEvents = relayEvents
+      this.latestScannedBlock = toBlock
+    }
+
+    if (eventNames != null) {
+      const nameSet = new Set(eventNames)
+      relayEvents = relayEvents.filter(event => nameSet.has(event.event))
+    }
+
+    if (managers != null) {
+      const lowercaseSet = new Set(Array.from(managers).map(x => x.toLowerCase()))
+      relayEvents = relayEvents.filter(event => lowercaseSet?.has(event.returnValues.relayManager.toLowerCase()))
+    }
```

Comment:
You have an 'infrastructure' to only query these events. I don't think we are saving time by getting events we don't need and then keeping them for 10 blocks worth of time. I suggest we keep using log filters.